puppet/modules/jenkins_job_builder/manifests/config.pp

```diff
@@ -5,49 +5,30 @@
   String $username,
   String $password,
   Integer[0] $jenkins_jobs_update_timeout = 600,
-  String $command_arguments = 'foreman-infra-jenkins-job-update',
+  String $command_arguments = 'jenkins-job-update',
+  String $git_project_name = 'jenkins-jobs',
+  String $git_repo = 'https://github.com/theforeman/jenkins-jobs.git',
+  Optional[String] $git_ref = undef,
 ) {
   $config_name = $name
   $directory = '/etc/jenkins_jobs'
   $inifile = "${directory}/jenkins_jobs_${config_name}.ini"
 
-  file { "${directory}/${config_name}":
-    ensure  => directory,
-    owner   => 'root',
-    group   => 'root',
-    mode    => '0755',
-    recurse => true,
-    purge   => true,
-    force   => true,
-    source  => "puppet:///modules/jenkins_job_builder/${config_name}",
+  vcsrepo { "${directory}/${git_project_name}":
+    ensure   => latest,
+    provider => git,
+    source   => $git_repo,
+    revision => $git_ref,
     notify  => Exec["jenkins-jobs-update-${config_name}"],
   }
 
-  cron { "jenkins-jobs-update-${config_name}-delete-old":
-    ensure      => absent,
-    command     => "timeout 1h jenkins-jobs --conf ${inifile} update --delete-old ${directory}/${config_name} > /var/cache/jjb.xml",
-    hour        => 0,
-    minute      => 10,
-    environment => 'PATH=/bin:/usr/bin:/usr/sbin',
-    require     => File[$inifile],
-  }
-
   exec { "jenkins-jobs-update-${config_name}":
```

Comment:
Does it make sense to add `refreshonly => true` so it only runs i the `vcsrepo` resource changes?